src/NServiceBus.Core/MessageInterfaces/MessageMapper/Reflection/MessageMapper.cs

```diff
@@ -150,6 +150,14 @@ void InitType(Type t)
                 return;
             }
 
+            var typeName = GetTypeName(t);
+
+            if (!t.IsInterface)
+            {
+                nameToType[typeName] = t.TypeHandle;
```

Comment:
Bailing out early here does mean that for concrete types, we're never populating the `typeToConstructor` collection with a constructor, so `CreateInstance` always has to fall back to calling `FormatterServices.GetUninitializedObject`

What if we instead wrap the field/property InitType calls at the end of the method in a `if (t.IsInterface)` check?